### ISSUE TITLE
The Input Image Normalization Policy used in your pretrained backbone network

### ISSUE THREAD

**Issue by developer0hye** [yjh0410/pytorch-yolo-v2#4]
Thanks for sharing great works!

I have one question.

What is your input image normalization policy used in pretrained backbone network?

Did you put the image into model without normalization?

Is the range  [0, 255]? or you followed the normalization method proposed in VGG(Z-normalization using mean and std)?

**Comment by yjh0410** [comment]
Hi ! Nice to meet you !
I use ToTensor to normalize the input image. But during traing YOLO, I forgot to normlize the input image......only (input - means) . For more details about training dartnet-19 and darknet-53, I suggest you to see my another project https://github.com/yjh0410/pytorch-imagenet. 
As for pretrained model file, I'm try to upload them to google drive.
Thanks for your help !

**Comment by developer0hye** [comment]
@yjh0410 
Thank you!!

**Comment by developer0hye** [comment]
@yjh0410 
If we re-train the yolov3 model with the normalization method used in a pre-training step, can we get the improvement? what do you think of this?

**Comment by yjh0410** [comment]
I tried. After data-augmentation, I divided it by 255, but it didn't bring any improvement. I got the almost same result.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"Yonghye Kwon"<notifications@github.com&gt;;
发送时间:&nbsp;2020年3月13日(星期五) 中午11:25
收件人:&nbsp;"yjh0410/pytorch-yolo-v2"<pytorch-yolo-v2@noreply.github.com&gt;;
抄送:&nbsp;"ら ．Secret"<1394571815@qq.com&gt;;"Mention"<mention@noreply.github.com&gt;;
主题:&nbsp;Re: [yjh0410/pytorch-yolo-v2] The Input Image Normalization Policy used in your pretrained backbone network (#4)





 
@yjh0410
 If we re-train the yolov3 model with the normalization method used in a pre-training step, can we get the improvement? what do you think of this?

**Comment by developer0hye** [comment]
@yjh0410 
How do you modify the code? 

To modify the code correctly, I think that we should follow below rules.

1. Do Not Subtract Mean during SSD Augmentation

2. Divided image by 255 after SSD Augmentation

3. Convert opencv bgr img([0, 1] to pytorch bgr format

4. Apply the normalization policy that we applied the backbone during pre-training.

Did you follow the above step???

**Comment by yjh0410** [comment]
In my data-augmentation code, its augmentation rules is as following:
1. it fisrt convert opencv BGR img into RGB format ( images = images[:, :, (2, 1, 0)]).
2. SSD Augmentation;
3. After step 2, it resize images and substract mean.
4. Finally, it convert opencv images into a torch.tensor type.


For more details, you could open utils/augmentation to check.


Besides those above, I added an extra step:
5. divide images by 255.


But I found there was no obvious improvement with step 5. So I removed it (Maybe I should keep it since it won't impair the performance).


"Apply the normalization policy that we applied the backbone during pre-training."


I'm sorry, I didn't follow this step.


I believe that there is a lot of room for improvement to make the model better and stronger. If I have enough time, I will try again.
------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"Yonghye Kwon"<notifications@github.com&gt;;
发送时间:&nbsp;2020年3月13日(星期五) 晚上9:13
收件人:&nbsp;"yjh0410/pytorch-yolo-v2"<pytorch-yolo-v2@noreply.github.com&gt;;
抄送:&nbsp;"ら ．Secret"<1394571815@qq.com&gt;;"Mention"<mention@noreply.github.com&gt;;
主题:&nbsp;Re: [yjh0410/pytorch-yolo-v2] The Input Image Normalization Policy used in your pretrained backbone network (#4)





 
@yjh0410
 How do you modify the code?
 
To modify the code correctly, I think that we should follow below rules.
  
 
Do Not Subtract Mean during SSD Augmentation
 
 
 
Divided image by 255 after SSD Augmentation
 
 
 
Convert opencv bgr img([0, 1] to pytorch bgr format
 
 
 
Apply the normalization policy that we applied the backbone during pre-training.
 
  
Did you follow the above step???

**Comment by developer0hye** [comment]
Thanks!
